content/blog/2020-08-19-train-a-sequence-model-with-poutyne/index.en.md

```diff
@@ -0,0 +1,613 @@
+dataset_vectorizer.vectorize(gb_data)
+dataset_vectorizer.vectorize(ru_data)
+dataset_vectorizer.vectorize(mx_data)
+```
+
+##### First Test
+
+Now let's test for the United States of America and United Kingdom.
+
+```python
+us_loader = DataLoader(us_data, batch_size=batch_size, collate_fn=pad_collate_fn)
+exp.test(us_loader)
+exp_bi_lstm.test(us_loader)
+
+gb_loader = DataLoader(gb_data, batch_size=batch_size, collate_fn=pad_collate_fn)
+exp.test(gb_loader)
+exp_bi_lstm.test(gb_loader)
+```
+
+The next table presents the results of both models for both countries. We obtain
+better results for the two countries using the bidirectional bi-LSTM (around 8% better). It's interesting to see that considering that their structures are similar to those in the training dataset (Canada), we obtain near as good results as those observed during training. Meaning that our model seems to have learned to recognize the structure of an address. Also, the
+presence of the same language as in the training dataset (i.e. English), we obtain poorer results than before. That situation
+is mostly due to the fact that the postal code formats are not the same. For the US, it is 5 digits, and for the UK it is similar to that of Canada, but it is not always a letter followed by a number, and it is not always 6 characters. It's *normal* for a model to
+have difficulty when faced with new patterns. All in all, we can say that our model has achieved good results.
+
+| Model (Country) | LSTM one layer | Bidirectional bi-LSTM |
+|:---------------:|:--------------:|:------------------:|
+|    Loss (US)    |     0.6176     |       **0.3078**   |
+|  Accuracy (US)  |     84.7396    |       **91.8220**  |
+|    Loss (UK)    |     0.4368     |       **0.1571**   |
+|  Accuracy (UK)  |     86.2543    |       **95.6840**  |
+
+
+##### The Second and Third Test
+
+Now let's tests for Russia and Mexico.
+
+```python
+ru_loader = DataLoader(ru_data, batch_size=batch_size, collate_fn=pad_collate_fn)
+exp.test(ru_loader)
+exp_bi_lstm.test(ru_loader)
+
+mx_loader = DataLoader(mx_data, batch_size=batch_size, collate_fn=pad_collate_fn)
+exp.test(mx_loader)
+exp_bi_lstm.test(mx_loader)
+```
+
+The next table presents the results of both models for the two countries tested. We see that the first test 
+(RU) gives poorer results than those for Mexican addresses, even if these latter are written in a different structure and language. 
+This situation could be explained by both languages' roots; Spanish is closer to French than Russian is. 
+An interesting thing is that even in a *difficult* annotation context, both models perform relatively well. 
+It suggests that our models have really learned the *logic* of an address sequence. It could also mean that if 
+we train our model longer, we could potentially improve our results. Other modifications that could improve our models are discussed in the next and final section.
+
+| Model (Country) | LSTM one layer | Bidirectional bi-LSTM |
```

Comment:
Don't you wish to address the simpler model being better on Russian than the larger one ?